cloudify_agent/api/pm/base.py

```diff
@@ -168,108 +161,88 @@ class Daemon(object):
     # override this when adding implementations.
     PROCESS_MANAGEMENT = None
 
-    # add specific mandatory parameters for different implementations.
-    # they will be validated upon daemon creation
-    MANDATORY_PARAMS = ['rest_host', 'broker_ip', 'local_rest_cert_file']
-
-    def __init__(self, logger=None, **params):
-
-        """
-
-        ####################################################################
-        # When subclassing this, do not implement any logic inside the
-        # constructor except for in-memory calculations and settings, as the
-        # daemon may be instantiated many times for an existing agent. Also,
-        # all daemon attributes must be JSON serializable, as daemons are
-        # represented as dictionaries and stored as JSON files on Disk. If
-        # you wish to have a non serializable attribute, mark it private by
-        # naming it _<name>. Attributes starting with underscore will be
-        # omitted when serializing the object.
-        ####################################################################
-
-        :param logger: a logger to be used to log various subsequent
-        operations.
-        :type logger: logging.Logger
-
-        :param params: key-value pairs as stated above.
-        :type params dict
-
-        """
-
-        # will be populated later on with runtime properties of the host
-        # node instance this agent is dedicated for (if needed)
-        self._runtime_properties = None
-
-        # configure logger
+    def __init__(
+        self,
+        *,
+        logger=None,
```

Comment:
Might also be worth type-hinting them?

Reply:
(with the validation removed that'd have some more benefit I suspect)

Reply:
eh, not that useful. It's all strings...